RecoTracker/MkFit/plugins/createPhase1TrackerGeometry.cc

```diff
@@ -0,0 +1,372 @@
+    {
+      ip.nlayers_per_seed = 3;
+      ip.maxCandsPerSeed = 2;
+      ip.maxHolesPerCand = 4;
+      ip.maxConsecHoles = 1;
+      ip.chi2Cut = 30;
+      ip.chi2CutOverlap = 3.5;
+      ip.pTCutOverlap = 1;
+    } else if (it == 7)  // for PixelLess step, maxCandsPerSeed=2 and maxHolesPerCand=maxConsecHoles=0
+    {
+      ip.nlayers_per_seed = 3;
+      ip.maxCandsPerSeed = 2;
+      ip.maxHolesPerCand = 0;
+      ip.maxConsecHoles = 0;
+      ip.chi2Cut = 30;
+      ip.chi2CutOverlap = 3.5;
+      ip.pTCutOverlap = 1;
+    } else if (it == 8)  // for TobTec step, maxCandsPerSeed=2 and maxHolesPerCand=maxConsecHoles=0
+    {
+      ip.nlayers_per_seed = 3;
+      ip.maxCandsPerSeed = 2;
+      ip.maxHolesPerCand = 0;
+      ip.maxConsecHoles = 0;
+      ip.chi2Cut = 30;
+      ip.chi2CutOverlap = 3.5;
+      ip.pTCutOverlap = 1;
+    }
+  }
+
+  void fillHitSelectionWindowsParams(IterationConfig &ic) {
+    HitSelectionWindowsPhase1 hsw;
+    for (int l = 0; l < (int)ic.m_layer_configs.size(); ++l) {
+      // dphi cut
+      ic.m_layer_configs[l].c_dp_0 = hsw.m_dp_params[ic.m_iteration_index][l][0];
+      ic.m_layer_configs[l].c_dp_1 = hsw.m_dp_params[ic.m_iteration_index][l][1];
+      ic.m_layer_configs[l].c_dp_2 = hsw.m_dp_params[ic.m_iteration_index][l][2];
+      // dq cut
+      ic.m_layer_configs[l].c_dq_0 = hsw.m_dq_params[ic.m_iteration_index][l][0];
+      ic.m_layer_configs[l].c_dq_1 = hsw.m_dq_params[ic.m_iteration_index][l][1];
+      ic.m_layer_configs[l].c_dq_2 = hsw.m_dq_params[ic.m_iteration_index][l][2];
+      // chi2 cut (for future optimization)
+      ic.m_layer_configs[l].c_c2_0 = hsw.m_c2_params[ic.m_iteration_index][l][0];
+      ic.m_layer_configs[l].c_c2_1 = hsw.m_c2_params[ic.m_iteration_index][l][1];
+      ic.m_layer_configs[l].c_c2_2 = hsw.m_c2_params[ic.m_iteration_index][l][2];
+    }
+  }
+
+  void PartitionSeeds0(const TrackerInfo &trk_info,
+                       const TrackVec &in_seeds,
+                       const EventOfHits &eoh,
+                       IterationSeedPartition &part) {
+    // Seeds are placed into eta regions and sorted on region + eta.
+
+    const int size = in_seeds.size();
+
+    for (int i = 0; i < size; ++i) {
+      const Track &S = in_seeds[i];
+
+      const bool z_dir_pos = S.pz() > 0;
+
+      HitOnTrack hot = S.getLastHitOnTrack();
+      // MIMI ACHTUNG -- here we assume seed hits have already been remapped.
+      // This was true at that time :)
+      float eta = eoh[hot.layer].GetHit(hot.index).eta();
+      // float  eta = S.momEta();
+
+      // Region to be defined by propagation / intersection tests
+      TrackerInfo::EtaRegion reg;
+
+      // Hardcoded for cms ... needs some lists of layers (hit/miss) for brl / ecp tests.
+      // MM: Check lambda functions/std::function
+      const LayerInfo &outer_brl = trk_info.outer_barrel_layer();
+
+      const LayerInfo &tib1 = trk_info.m_layers[4];
```

Comment:
wouldn't it better to have a sort of enum for handling this index ?
ok, the geometry (above all the Phase1 one is well established ....)

Reply:
An enumeration would not suit too well because the (mkFit internal) layer numbering depends on the geometry. E.g. phase2 (or phase0) would have different layer numbers for the layers/disks of "TIB1" or "TEC1+". Therefore it would be better for these constants to be defined at run time (somewhat similar to `TrackerTopology`). We could e.g.
* have `createPhase1TrackerGeometryAutoGen()` to fill an auxiliary struct holding that information, or
* extend `mkfit::LayerNumberConverter` that converts CMSSW layer numbers to mkFit layer numbers (that should also eventually be absorbed in CMSSW as well) to give this information (would need a bit more work than the previous option)

Reply:
Yes. Also note that this is geometry specific function that will likely be very different for newer geometries, potentially introducing two transition regions to cover different trajectories through the sub-detectors.